fsharp-backend/src/LibExecution/Interpreter.fs

```diff
@@ -203,17 +214,17 @@ let rec eval' (state : ExecutionState) (st : Symtable) (e : Expr) : DvalTask =
         // live values.
         let fakeST = Map.add "var" (DIncomplete SourceNone) st
         do! preview fakeST body
+
       let parameters =
-        List.choose
-          (function
+        parameters
+        |> List.choose (function
           | _, "" -> None
           | id, name -> Some(id, name))
-          parameters
 
       // It is the responsibility of wherever executes the DBlock to pass in
       // args and execute the body.
       return DFnVal(Lambda { symtable = st; parameters = parameters; body = body })
-    | EMatch (id, matchExpr, cases) ->
+    | EMatch (id, matchExpr, caseExprs) ->
```

Comment:
I think "cases" is a better name here. The values in caseExprs are not expressions, they're a list of pattern+expr.